[SME][compiler-rt] Guard .variant_pcs under HAS_ASM_SME macro.

sdesmalen-arm opened the pull request:
This also adds the .variant_pcs directive to some functions from which it was previously missing.

sdesmalen-arm committed:
[SME][compiler-rt] Guard .variant_pcs under HAS_ASM_SME macro.

This also adds the .variant_pcs directive to some functions from
which it was previously missing.

paulwalker-arm reviewed (commented):
If `.variant_pcs` is optional, why do we need it at all?

sdesmalen-arm committed:
Disable SME if asm support not available [to squash]

This is mostly needed because the .variant_pcs directive is required to
build the SME ABI routines, but takes it one step further by requiring
SME assembler support from the compiler. This removes the need for the
macros that use .inst <encoding> otherwise.

sdesmalen-arm commented:
> If `.variant_pcs` is optional, why do we need it at all?

AIUI `.variant_pcs` is not optional. I've removed the support to build these routines when SME ASM support is not available, which also gets rid of some of the macros and cleans up the code a bit further.
